pkg/services/settings/settingsprovider/settingsprovider.go

```diff
@@ -0,0 +1,162 @@
+package settingsprovider
+
+import (
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/registry"
+	"github.com/grafana/grafana/pkg/services/settings"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+var (
+	logger = log.New("settingsprovider")
+)
+
+func init() {
+	registry.Register(&registry.Descriptor{
+		Name:         "SettingsProvider",
+		Instance:     &Implementation{},
+		InitPriority: registry.High - 5,
+	})
+}
+
+type settingsBag map[string]map[string]string
+
+type Implementation struct {
+	sync.RWMutex
+	FileCfg  *setting.Cfg       `inject:""`
+	SQLStore *sqlstore.SQLStore `inject:""`
+	settings settingsBag
+}
+
+func (i *Implementation) Init() (err error) {
+	i.settings, err = i.loadAndMergeSettings()
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		syncTicker := time.NewTicker(1 * time.Minute)
+		for {
+			<-syncTicker.C
+			logger.Info("Checking for new updates")
+
+			err := i.Refresh()
+			if err != nil {
+				logger.Error("Error while refreshing settings", "error", err.Error())
+			}
+		}
+	}()
+
+	return
+}
+
+func (i *Implementation) Section(name string) settings.Section {
```

Comment:
instead of building a static view of the section at the point in time this method is called, you should just return a struct that embeds the section name and a reference to the provider Implementation.
This will ensure that all future calls to the returned Section's `KeyValue(key string)` method will return current data.

eg
```
type settingSection struct {
	section string
	impl    *Implementation
}

func (s settingSection) KeyValue(key string) settings.KeyValue {
	return s.impl.KeyValue(s.section, key)
}

func (i *Implementation) Section(name string) settings.Section {
	return settingSection{impl: i, section: name}
}
```

Reply:
Yeah, it definitely makes much more sense, and it addresses the aforementioned concern 👌🏻  
Thanks for the suggestion!! 🙏🏻